Source/dom/components/Library.js

```diff
@@ -0,0 +1,145 @@
+import { WrappedObject, DefinedPropertiesKey } from '../WrappedObject'
+import { Document } from './Document'
+import { toArray, getURLFromPath, getDocumentData } from '../utils'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { wrapObject } from '../wrapNativeObject'
+import { ImportableObject } from './ImportableObject'
+
+const AddStatus = {
+  0: 'ok',
+  1: 'the library has already been added',
+  2: 'the document is not in the new JSON format',
+  3: 'there was a problem reading the asset library file',
+}
+
+/**
+ * A Sketch Library.
+ */
+export class Library extends WrappedObject {
+  constructor(library = {}) {
+    if (!library.sketchObject) {
+      throw new Error('Cannot create a new Library directly')
+    }
+
+    super(library)
+  }
+
+  static getLibraries() {
+    const libraryController = AppController.sharedInstance().librariesController()
+    return toArray(libraryController.libraries()).map(
+      Library.fromNative.bind(Library)
+    )
+  }
+
+  static getLibraryForDocumentAtPath(path) {
+    const libUrl = getURLFromPath(path)
+
+    const libraryController = AppController.sharedInstance().librariesController()
+
+    // check if we already imported the library
+    const existingLibraries = libraryController.libraries()
+    for (let i = 0; i < existingLibraries.count(); i += 1) {
+      const existingLibrary = existingLibraries.objectAtIndex(i)
+      if (existingLibrary.locationOnDisk().isEqual(libUrl)) {
+        return Library.fromNative(existingLibrary)
+      }
+    }
+
+    // otherwise, let's add it
+    const status = libraryController.addAssetLibraryAtURL(libUrl)
+
+    if (status !== 0) {
+      throw new Error(`Error while adding the library: ${AddStatus[status]}.`)
+    }
+
+    const lib = libraryController.userLibraries().firstObject()
```

Comment:
is it good to rely on `userLibraries` being ordered by most recently added?

Reply:
Yep, see previous code review :smile:

Reply:
gotcha 👍